Ajustement on the behavior of the stock mvt (#302)
PrestaShop/prestashop-specs: content/1.7/back-office/catalog/stock/stock.md

---
title: Stock
weight: 1
---

  
# SPECS Stock & Stock movement

## Stock

  
| Stock's type | Definition | When impacted | 
|-----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Physical stock | Number of products that are physically in my stock.| When a merchant can add or remove physical stock 

Reviewer: ```suggestion
| Physical stock | Number of products that are physically in my stock.| When a merchant can add or remove physical stock, when an order is shipped |
```

Author: Not really, it is a merchant edit the physical stock so when it is shipped and when he increases or decrease it from the product page, stock page or from another source

> | Physical stock | Number of products that are physically in my stock.| When a merchant can add or remove physical stock 

Reviewer 2: IMHO we can't list every use case next to the definition, but user stories should provide the details.
```suggestion
| Physical stock | Number of products that are physically in my stock.| When a merchant can add or remove physical stock, when ordered products leave or return in stock, ...
```

> | Reserved stock | Number of products that are part of customer orders that aren’t shipped yet.| When client's merchant order products |

Reviewer: ```suggestion
| Reserved stock | Number of products that are part of customer orders that aren’t shipped yet.| When customers order products |
```

| Available stock | Number of products that are available for sale. Available stock = physical stock - reserved stock | When Physical and Reserved Stock are impacted

## Stock movement

**User story**

 As a merchant, I want to associate a stock edition with a movement type to track of the reason of a given physical stock movement.

  Stock movement | When the stock movement is saved | Impact
|:--------:| :-------------|:-------------|

> Product return | When a user re-stocks a product from an Order that has been flagged as _shipped_ | The quantity re-stocked increases the product's physical stock | 

Reviewer: ```suggestion
Product return | When a user re-stocks a product from an Order that has been flagged as _shipped_ | The quantity re-stocked increases the product's physical (and available) stock | 
```

Author: I will do a matrix as suggested, it will be more explicit

Reviewer 2: @jolelievre the impact on available stock should be implicit by definition of "available", but it doesn't hurt to remind the impact here
```suggestion
Product return | When a merchant restocks a product from an Order that has been flagged as _shipped_ | The restocked quantity increases the product's physical stock, and implicitly its available stock | 
```

> Employee edition | When the physical stock is edited without being linked to a customer order | The physical stock is increased or decreased

Reviewer 2: ```suggestion
Employee edition | When the merchant edits the physical stock from the BO, without any link to a customer order | The physical stock is increased or decreased
```
